src/commands/contract/test/index.ts

```diff
@@ -0,0 +1,182 @@
+import "ts-mocha";
+import { Flags, Args } from "@oclif/core";
+import path from "node:path";
+import { globby } from "globby";
+import Mocha from "mocha";
+import { emptyDir } from "fs-extra/esm";
+import shell from "shelljs";
+import { Contract } from "../../../lib/contract.js";
+import { SwankyCommand } from "../../../lib/swankyCommand.js";
+import { ConfigError, FileError, InputError, ProcessError, TestError } from "../../../lib/errors.js";
+import { spawn } from "node:child_process";
+import { Spinner } from "../../../lib/index.js";
+
+declare global {
+  var contractTypesPath: string; // eslint-disable-line no-var
+}
+
+export class TestContract extends SwankyCommand<typeof TestContract> {
+  static description = "Run tests for a given contact";
+
+  static flags = {
+    all: Flags.boolean({
+      default: false,
+      char: "a",
+      description: "Set all to true to compile all contracts",
+    }),
+    mocha: Flags.boolean({
+        default: false,
+        description: "Run tests with mocha",
+    }),
+  };
+
+  static args = {
+    contractName: Args.string({
+      name: "contractName",
+      default: "",
+      description: "Name of the contract to test",
+    }),
+  };
+
+  async run(): Promise<void> {
+    const { args, flags } = await this.parse(TestContract);
+
+    if (args.contractName === undefined && !flags.all) {
+      throw new InputError("No contracts were selected to compile");
+    }
+
+    const contractNames = flags.all
+      ? Object.keys(this.swankyConfig.contracts)
+      : [args.contractName];
+
+    const testDir = path.resolve("tests");
+
+    const spinner = new Spinner();
+
+    for (const contractName of contractNames) {
+      const contractRecord = this.swankyConfig.contracts[contractName];
+      if (!contractRecord) {
+        throw new ConfigError(
+          `Cannot find a contract named ${args.contractName} in swanky.config.json`
+        );
+      }
+
+      const contract = new Contract(contractRecord);
+
+      if (!(await contract.pathExists())) {
+        throw new FileError(
+          `Path to contract ${args.contractName} does not exist: ${contract.contractPath}`
+        );
+      }
+
+      console.log(`Testing contract: ${contractName}`);
+
+      if (!flags.mocha) {
+        await spinner.runCommand(
+          async () => {
+            return new Promise<string>((resolve, reject) => {
+              const compileArgs = [
+                "test",
```

Comment:
We should be able to run e2e tests against local swanky node by providing the following env variable: `CONTRACTS_NODE="LOCAL_SWANKY_NODE_PATH"`. This is optional and can be handle by an optional boolean `local-node`.

Reply:
I made it, but e2e-tests are not working on swanky-node for some reason